### PR TITLE
[DRAFT] : createCommit

### PR DESCRIPTION
**Pull request by ravishetye**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 1]
```diff
@@ -38,6 +39,27 @@ public Response setCommit(Commit commit, BlobFunction setBlobs, RepositoryFuncti
...
+      sb.append("parent:");
+      parentSHAs.forEach(pSHA -> sb.append(pSHA)); // TODO  EL/AJ to verify /optimize
+    }
+    sb.append(":message:")
+        .append(commit.getMessage())
+        .append(":date_created:")
+        .append(commit.getDateCreated())
+        .append(":author:")
+        .append(commit.getAuthor())
+        .append("blobs")
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: Blobs => folder? The commit just has a link to its root folder.

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 2]
```diff
@@ -38,6 +39,27 @@ public Response setCommit(Commit commit, BlobFunction setBlobs, RepositoryFuncti
...
+  private String generateCommitSHA(String blobSHA, Commit commit) throws NoSuchAlgorithmException {
+
+    StringBuilder sb = new StringBuilder();
+    if (!commit.getParentShasList().isEmpty()) {
+      List<String> parentSHAs = commit.getParentShasList();
+      Collections.sort(parentSHAs); // TODO  EL/AJ to verify /optimize
+      sb.append("parent:");
+      parentSHAs.forEach(pSHA -> sb.append(pSHA)); // TODO  EL/AJ to verify /optimize
+    }
+    sb.append(":message:")
```
Comment (ad-47) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: Here if `commit.getParentShasList()` is empty then this linr should be 
```suggestion
    sb.append("message:")
```

File: backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java [hunk 3]
```diff
@@ -87,50 +88,126 @@ InternalFolderElement saveFolders(Session session, FileHasher fileHasher)
...
+        break;
+      default:
+        break;
+    }
+  }
+
+  private String computeSHA(S3DatasetComponentBlob s3componentBlob)
+      throws NoSuchAlgorithmException {
+    StringBuilder sb = new StringBuilder();
+    sb.append(":path:").append(computeSHA(s3componentBlob.getPath()));
```
Comment (ad-47) [inline backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java]:
```suggestion
    sb.append("path:").append(computeSHA(s3componentBlob.getPath()));
```

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 1 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: yes verified.

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 2 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: does not matter we anyway use it to create SHA, I added the keys for debugging purposes

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 1 above)
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]:
```suggestion
        .append(":rootHash:")
```

File: backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java (hunk 3 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/DatasetComponentDAORdbImpl.java]: does not matter we anyway use it to create SHA, I added the keys for debugging purposes

**Pull request merged**